From 8b5fdc29cb2956cc91deb8b45a69661ebd180830 Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Wed, 17 Feb 2021 23:35:40 +0200
Subject: [PATCH] [SW-20154] gaudi2: Fix for overlapping DCCM region

There is an overlap in the DCCM memory region
for arcboot ROM and ppboot.

Current ppboot DCCM starts at : 0x80002000
Current Arcboot ROM DCCM usage :
0x80002028 + 0x2000 (size) = 0x80004028.

The offset for ppboot has been updated to maintain
unique memory access range in DCCM memory.

Change-Id: I01e68f4fe9746ba0e5df86b3e3a5a7f7c38808d8
Signed-off-by: Karthikeyan K <karthikk@habana.ai>
---

diff --git a/gaudi2_arc_gnu/common.h b/gaudi2_arc_gnu/common.h
index e421b96..4596652 100644
--- a/gaudi2_arc_gnu/common.h
+++ b/gaudi2_arc_gnu/common.h
@@ -33,10 +33,10 @@
 #define BOOT_INFO_BASE		SP_SRAM_BASE
 
 /* In order not to conflict with ARC BOOTROM's usage of DCCM,
- * Let PPBoot start from the offset of 8KB, for using DCCM
+ * Let PPBoot start from the offset of 0x402C, for using DCCM
  * for its own BSS and Stack.
  */
-#define PPBOOT_DCCM_BASE	0x80002000
+#define PPBOOT_DCCM_BASE	0x8000402C
 
 #define SDRAM_BASE		0x11000000
 #define FIT_IMG_BASE		0x11200000
